### PR TITLE
remove_op_node is part of the DAG API

### LINKED ISSUE #289: DAGCircuit Remove node
When modifying a DAGCircuit it is not easily possible to remove a specific node from the circuit. There exists a protected method `_remove_op_node` which does what I want to do. Is there a reason why this method needs to be protected?

## Expected Behavior
There should exist a way to remove a specific node from a DAGCircuit that does not use protected methods.

## Current Behavior
The only way currently is to either modify `multi_graph` directly (and risk breaking changes) or ignore the protected status and use `_remove_op_node` anyway (my choice currently).

## Possible Solution
Simple fix: Make the method public.
Complex fix: Rework the API for interacting with DAGCircuit. Currently, I am required to inspect `multi_graph` directly for many things that I wish to do with a DAGCircuit. I do not think this should be intended, since `multi_graph` encodes some fairly opaque internal state.

## Context
I am trying to transform a DAGCircuit for the [QISKit developer challenge](https://qx-awards.mybluemix.net/#qiskitDeveloperChallengeAward) to conform to a given hardware architecture. I do that by removing op nodes from the given DAGCircuit and building a DAGCircuit that conforms to the architecture.

## Your Environment
* Version used: qiskit 0.4.8
* Environment name and version (e.g. Python 3.6.1): python 3.6.4
* Operating System and version: macOS 10.13

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #289

The method `remove_op_node` should not be "private", so the underscore is removed (and a test is added). Welcome `remove_op_node` to the DAG API.

### CODE REVIEW

File: test/python/test_dagcircuit.py [hunk 1]
```diff
@@ -264,6 +264,16 @@ def test_dag_ops_on_wire(self):
...
+    def test_remove_op_node(self):
+        """ Test remove_op_node method."""
+        self.dag.apply_operation_back(HGate(self.qubit0))
+
+        op_nodes = self.dag.gate_nodes()
+        h_gate = op_nodes.pop()
+        self.dag.remove_op_node(h_gate)
+
+        self.assertEqual(len(self.dag.gate_nodes()), 0)
+
```
Comment (kdk) [inline test/python/test_dagcircuit.py]: Can you add a test for the case where we're removing a node from a larger graph? At least that we're removing the right node and that the remaining graph structure is correct (mainly that the re-wired edges are consistent).

File: qiskit/dagcircuit/dagcircuit.py [hunk 2]
```diff
@@ -1221,13 +1221,13 @@ def quantum_successors(self, node):
                 successors.append(successor)
         return successors
 
-    def _remove_op_node(self, node):
+    def remove_op_node(self, node):
```
Comment (kdk) [inline qiskit/dagcircuit/dagcircuit.py]: Should we check that `node` is of type `op`?

Comment (1ucian0) [reply qiskit/dagcircuit/dagcircuit.py]: True! Fixed in dc7631f8

File: test/python/test_dagcircuit.py (hunk 1 above)
Comment (1ucian0) [reply test/python/test_dagcircuit.py]: fixed in af2d8743

File: qiskit/dagcircuit/dagcircuit.py [hunk 3]
```diff
@@ -1221,17 +1221,21 @@ def quantum_successors(self, node):
...
         Add edges from predecessors to successors.
         """
         if isinstance(node, int):
-            warnings.warn('Calling _remove_op_node() with a node id is deprecated,'
+            warnings.warn('Calling remove_op_node() with a node id is deprecated,'
                           ' use a DAGNode instead',
                           DeprecationWarning, 2)
             node = self._id_to_node[node]
 
+        if node.type is not 'op':
```
Comment (kdk) [inline qiskit/dagcircuit/dagcircuit.py]:
```suggestion
        if node.type != 'op':
```

**Pull request merged**